scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -100,6 +116,77 @@ public void visit(ImplicitAnonymousFunctionExpressionNode implicitAnonymousFunct
         this.visitSyntaxNode(implicitAnonymousFunctionExpressionNode.expression());
     }
 
+    @Override
+    public void visit(ClassDefinitionNode classDefinitionNode) {
+        checkNonIsolatedPublicClassDefinitions(classDefinitionNode);
+        checkNonIsolatedPublicClassMembers(classDefinitionNode);
+    }
+
+    private void checkNonIsolatedPublicClassDefinitions(ClassDefinitionNode classDefinitionNode) {
+        semanticModel.symbol(classDefinitionNode).ifPresent(symbol -> {
+            if (symbol instanceof ClassSymbol classSymbol) {
+                List<Qualifier> qualifiers = classSymbol.qualifiers();
+                if (getQualifier(qualifiers, PUBLIC_KEYWORD.stringValue()) &&
+                        !getQualifier(qualifiers, ISOLATED_KEYWORD.stringValue())) {
+                    reportIssue(classDefinitionNode, CoreRule.PUBLIC_NON_ISOLATED_CONSTRUCT);
+                }
+            }
+        });
+    }
+
+    private void checkNonIsolatedPublicClassMembers(ClassDefinitionNode classDefinitionNode) {
+        classDefinitionNode.members().forEach(member -> {
+            semanticModel.symbol(member).ifPresent(symbol -> {
+                if (symbol.kind() == SymbolKind.METHOD) {
```

Comment:
This doesn't cover resources, does it?

Reply:
For resources, the language itself gives a warning if it is not isolated.
So do we need to include it here also?

Reply:
Had a offline chat with @MaryamZi and decided that we don not need warnings for resources, instead of that add warnings for objects